### PR TITLE
Allow distributions to be absent in deserialization

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/uv/issues/5434.

### CODE REVIEW

File: crates/uv/tests/sync.rs [hunk 1]
```diff
@@ -162,3 +162,46 @@ fn frozen() -> Result<()> {
...
+
+    // Running `uv sync` should generate an empty lockfile.
+    uv_snapshot!(context.filters(), context.sync(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: `uv sync` is experimental and may change without warning
+    warning: No `requires-python` value found in the workspace. Defaulting to `>=3.12`.
```
Comment (zanieb) [inline crates/uv/tests/sync.rs:186]: Maybe we should throw a requires-python in there to reduce the noise?

**Review by zanieb — APPROVED** [review]

File: crates/uv/tests/sync.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv/tests/sync.rs:186]: Not sure if we "can" since it's a virtual workspace (not a project). I'll try.

Comment (charliermarsh) [reply crates/uv/tests/sync.rs:186]: Typically it would have at least one member.

Comment (charliermarsh) [reply crates/uv/tests/sync.rs:186]: Yeah as soon as we add a `project` field it becomes non-virtual and the package itself gets included in the lockfile (making it non-empty).

**Pull request merged**